<launch>

<!-- Arguments -->

    <!--Pocs matches -->
    <!-- <arg name = "bagfile" default = "/home/miguel/Desktop/bagfiles/2020_07_23_cala_lliteres/15_51_42/images_2020-07-23-15-51-43_0.bag" />  -->
    <!--Va bastant bÃ© -->
    <arg name = "bagfile" default = "/home/miguel/Desktop/bagfiles/2020_09_18/14_16_55/images_2020-09-18-14-16-57_0.bag" /> 
    <!-- Error al inicialitzar -->
    <!-- <arg name = "bagfile" default = "/home/miguel/Desktop/bagfiles/2020_10_09/14_03_13/images_2020-10-09-14-03-14_0.bag" />  -->
    <!-- Error al inicialitzar -->
    <!-- <arg name = "bagfile" default = "/home/miguel/Desktop/bagfiles/2020_09_17/15_19_55/images_2020-09-17-15-19-57_0.bag" /> --> 

    <arg name = "camera" default = "/stereo_down"/>
    <param name = "/use_sim_time" value = "true"/>

    <!-- Play the bagfile -r 0.5 -->
    <node pkg = "rosbag" type = "play" name = "rosbag" args = "--clock $(arg bagfile)" />

    <!-- Run the stereo image proc -->
    <node ns = "$(arg camera)" pkg = "stereo_image_proc" type = "stereo_image_proc" name = "stereo_image_proc" respawn = "true" output = "screen">
        <!-- <rosparam file="/home/uib/turbot_ws/src/turbot_stack/cola2_turbot/config/avt_manta/disparity_params_x1.yaml"/> -->
        <param name = "approximate_sync" value = "True" />
        <param name = "queue_size" value = "100" />
    </node>

    <!-- Test SIFT-->
    <node pkg = "viso2_ros" type = "testSIFT" name = "testSIFT" output = "screen">
        <remap from = "stereo" to = "$(arg camera)"/>
        <!-- <remap from="image" to="image_rect"/> -->
        <!-- <param name = "base_link_frame_id" value = "turbot/base_link"/> -->
        <param name = "INFO_MATCHING" value="true" />
    </node>

</launch>